src/Datasource/Query.php

```diff
@@ -199,7 +495,7 @@ public function getEndpoint(): Endpoint
      * @param \Muffin\Webservice\Webservice\WebserviceInterface $webservice The webservice to use
      * @return $this
      */
-    public function setWebservice(WebserviceInterface $webservice)
+    public function setWebservice(WebserviceInterface $webservice): Query
```

Comment:
```suggestion
    public function setWebservice(WebserviceInterface $webservice)
```

Functions returning `$this` shouldn't have a return type.

Reply:
Me, psalm and phpstorm don't agree with that. Anyway, there are now no functions that return $this anymore anyway. They now all return self, because psalm presribes that.